lib/identity_cache/memoized_cache_proxy.rb

```diff
@@ -79,33 +104,56 @@ def fetch(key)
     def fetch_multi(*keys)
       memoized_keys, missed_keys = [], [] if IdentityCache.logger.debug?
 
-      result = if memoizing?
-        hash = {}
-        mkv = memoized_key_values
-
-        non_memoized_keys = keys.reject do |key|
-          if mkv.has_key?(key)
-            memoized_keys << key if IdentityCache.logger.debug?
-            hit = mkv[key]
-            hash[key] = hit unless hit.nil?
-            true
+      miss_count = 0
+      memoized_count = 0
+
+      memoizing = memoizing?
+      result = ActiveSupport::Notifications.instrument('identity_cache.cache.fetch_multi', { memoizing: memoizing }) do |payload|
+        payload[:time_in_sor] = 0
+        result = if memoizing
```

Comment:
Similar to my suggestion to extract out this branch on `fetch`, I think we can do the same here, where we can extract out even more logic

```ruby
    def fetch_multi_memoized(keys)
      return yield keys unless memoizing?

      result = {}
      missing_keys = keys.reject do |key|
        if memoized_key_values.key?(key)
          result[key] = memoized_key_values[key]
          true
        end
      end

      unless missing_keys.empty?
        block_result = yield missing_keys
        memoized_key_values.merge!(block_result)
        result.merge!(block_result)
      end

      result
    end
```